native/acter/src/api/push/notification_item.rs

```diff
@@ -926,552 +450,116 @@ impl NotificationItem {
-                    .and_then(|o| NotificationItemParent::try_from(&o).ok());
-
-                if let Some(new_title) = e.content.name {
-                    Ok(builder
-                        .title(new_title)
-                        .inner(NotificationItemInner::TitleChange {
-                            parent_obj,
-                            parent_id: e.content.task_list.event_id,
-                            room_id: e.room_id,
-                            event_id: e.event_id,
-                        })
-                        .build()?)
-                } else if let Some(Some(new_content)) = e.content.description {
-                    return Ok(builder
-                        .inner(NotificationItemInner::DescriptionChange {
-                            parent_obj,
-                            parent_id: e.content.task_list.event_id,
-                            room_id: e.room_id,
-                            event_id: e.event_id,
-                            content: Some(new_content),
-                        })
-                        .build()?);
-                } else {
-                    // fallback: other changes
-                    return Ok(builder
-                        .inner(NotificationItemInner::OtherChanges {
-                            parent_obj,
-                            parent_id: e.content.task_list.event_id,
-                            room_id: e.room_id,
-                            event_id: e.event_id,
-                        })
-                        .build()?);
-                }
-            }
-            // -- Task Specific
-            AnyActerEvent::Task(MessageLikeEvent::Original(e)) => {
-                let parent_obj = client
-                    .store()
-                    .get(&e.content.task_list_id.event_id)
-                    .await
-                    .map_err(|error| {
-                        tracing::error!(?error, "Error loading parent of comment");
-                    })
-                    .ok()
-                    .and_then(|o| NotificationItemParent::try_from(&o).ok());
-
-                Ok(builder
-                    .inner(NotificationItemInner::TaskAdd {
-                        parent_obj,
-                        parent_id: e.content.task_list_id.event_id,
-                        room_id: e.room_id,
-                        event_id: e.event_id,
-                    })
-                    .title(e.content.title)
-                    .build()?)
-            }
-            AnyActerEvent::TaskUpdate(MessageLikeEvent::Original(e)) => {
-                let parent_obj = client
-                    .store()
-                    .get(&e.content.task.event_id)
-                    .await
-                    .map_err(|error| {
-                        tracing::error!(?error, "Error loading parent of comment");
+        let mut builder = self;
+
+        // a few special cases we want to deal with
+        let builder = match activity.content() {
```

Comment:
Notifications have a peculiar pattern and we need to wrap some of those inner types into the specific patterns -- most notably to have the right `title` set on the notification itself. This is what this match here does.

There are e2e tests for all of these.